addon/components/base/bs-nav/item.js

```diff
@@ -64,10 +85,29 @@ export default Component.extend(ComponentParent, {
    * @type boolean
    * @public
    */
-  active: overrideableCP('_active', 'linkToParams.[]', 'router.currentURL', function() {
-    let params = this.get('linkToParams');
+  active: overrideableCP('_active', 'route', 'model', 'models', 'query', 'router.currentURL', function() {
+    let { route, model, models, query } = this.getProperties('route', 'model', 'models', 'query');
+    let params = [];
+
+    assert('You cannot pass both `@model` and `@models` to a nav item component!', !model || !models);
```

Comment:
I would not have expected this assertion to be part of the `active` computed property. It's a general requirement not only related to active state of a link. Did you considered moving it to `init`?

Reply:
Makes sense!